src/Emulator.hpp

```diff
@@ -14,14 +14,37 @@ class Emulator
 private:
     bool Initialize();
 
+    struct SDLWindowDeleter
+    {
+        void operator()(SDL_Window* window)
+        {
+            if (window != nullptr)
+            {
+                SDL_DestroyWindow(window);
+                Logger::Log("SDL_Window destroyed.");
+            }
+        }
+    };
+
+    struct SDLRendererDeleter
+    {
+        void operator()(SDL_Renderer* renderer)
+        {
+            if (renderer != nullptr)
+            {
+                SDL_DestroyRenderer(renderer);
+                Logger::Log("SDL_Renderer destroyed.");
+            }
+        }
+    };
+
```

Comment:
The SDLWindowDeleter and SDLRendererDeleter is how you tell unique_ptr what to do when the object is to be cleaned up.  I added logs to show you that it does indeed work!